### PR TITLE
[AIRFLOW-5434] Use hook to provide credentials in GKEPodOperator

### PR DESCRIPTION
**Pull request by mik-laj**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5434
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Comment by mik-laj** [comment]
I am wondering about the optional ``gcp_conn_id`` parameter. It was optional. Now it's required.  When the user did not provide a value, credentials was set up to use ADC strategies. I think we should abandon this mechanism for security reasons. Credentials should only be determined based on the connection configuration.  Now this is not a big problem, but in the future this connection configuration only by connecting can be crucial. In future, it may allow the introduction of further security mechanisms, e.g. IAM.  

Do you think it is worth introducing backward compatibility or leaving the current implementation? If we leave the current implementation then I have to add a note in UPDATING.md

It is worth noting that ADC is quite a complex mechanism and permissions may be mistakenly granted.

File: airflow/contrib/hooks/gcp_api_base_hook.py [hunk 1]
```diff
@@ -283,32 +284,50 @@ def inner_wrapper(self: GoogleCloudBaseHook, *args, **kwargs) -> RT:
     @staticmethod
     def provide_gcp_credential_file(func: Callable[..., RT]) -> Callable[..., RT]:
         """
-        Function decorator that provides a ``GOOGLE_APPLICATION_CREDENTIALS``
-        environment variable, pointing to file path of a JSON file of service
-        account key.
+        Function decorator that provides a GCP credentials for application supporting Application
```
Comment (mik-laj) [inline airflow/contrib/hooks/gcp_api_base_hook.py]: i think, we can use https://pypi.org/project/contextdecorator/ here

**Comment by potiuk** [comment]
@mik-laj  - Also Travis is sad...

**Comment by mik-laj** [comment]
Can you answer an open question? https://github.com/apache/airflow/pull/6050#issuecomment-529115287

**Comment by potiuk** [comment]
Using gcp_conn_id is preferred way. You can update UPDATING.md and forget about backwards compatibility.

**Review by potiuk — APPROVED** [review]

**Pull request merged**